theatre/studio/src/uiComponents/usePresence.tsx

```diff
@@ -0,0 +1,231 @@
+import type {StudioSheetItemKey} from '@theatre/shared/utils/ids'
+import type {StrictRecord} from '@theatre/shared/utils/types'
+import React, {useMemo} from 'react'
+import {useEffect} from 'react'
+import {useLogger} from './useLogger'
+import {Box, prism, valueDerivation} from '@theatre/dataverse'
+import {Atom} from '@theatre/dataverse'
+import {useVal} from '@theatre/react'
+
+export enum Hovered {
+  /** This is the focus */
+  Hovered,
+  /** The related parent */
+  Parent,
+  /** The related child / moves with */
+  Child,
+  /** The object containing the hovered item */
+  ParentObject,
+}
+
+function createPresence(): Presence {
+  const currentPresence = new Box<StudioSheetItemKey | undefined>(undefined)
+  // TODO: move into ephemeral
+  const atom = new Atom({} as StrictRecord<StudioSheetItemKey, boolean>)
+
+  type HoveredWithImportance = {
+    importance: number
```

Comment:
how is `importance` used? I don't understand why I would need an importance for this.

Reply:
Added a comment like:
```
     * Weight to determine the precedence of each relationship (e.g. "Hovered" has a higher importance than "Sibling" if that were a thing)
     * As well, no matter the order with which relationships are added, the importance will determine the one shown.
```

Reply:
Now removed completely